src/views/examples/form-elements/index.njk

```diff
@@ -14,98 +25,156 @@
-    <label class="govuk-c-label" for="govuk-c-input-a">
-      National Insurance number
-    </label>
-    <input class="govuk-c-input" id="govuk-c-input-a" type="text">
-    <input class="govuk-c-button" type="submit" value="Save and continue">
-  </fieldset>
-
-  <fieldset class="govuk-c-fieldset" id="textarea">
-    <label class="govuk-c-label" for="govuk-c-textarea-a">
-      Why can't you provide a National Insurance number?
-    </label>
-    <textarea class="govuk-c-textarea" name="govuk-c-textarea-a" id="govuk-c-textarea-a" rows="5"></textarea>
-  </fieldset>
+  {% call govukFieldset() %}
+    <span id="textarea"></span>
+    {% from "textarea/macro.njk" import govukTextarea %}
+    {{ govukTextarea({
+      id: "govuk-c-textarea-a",
+      name: "govuk-c-textarea-a",
+      rows: "5",
+      label: {
+        text: "Why can't you provide a National Insurance number?"
+      }
+    }) }}
+  {% endcall %}
 
-  <fieldset class="govuk-c-fieldset" id="radio-stacked">
-    <legend class="govuk-c-fieldset__legend">
-      <h2 class="govuk-u-bold-24">How do you want to be contacted?</h2>
-    </legend>
-    <div class="govuk-c-radio">
-      <input class="govuk-c-radio__input" id="radio-stacked-1" type="radio" name="radio-stacked-group" value="Email">
-      <label class="govuk-c-radio__label" for="radio-stacked-1">Email</label>
-    </div>
-    <div class="govuk-c-radio">
-      <input class="govuk-c-radio__input" id="radio-stacked-2" type="radio" name="radio-stacked-group" value="Phone">
-      <label class="govuk-c-radio__label" for="radio-stacked-2">Phone</label>
-    </div>
-    <div class="govuk-c-radio">
-      <input class="govuk-c-radio__input" id="radio-stacked-3" type="radio" name="radio-stacked-group" value="Text">
-      <label class="govuk-c-radio__label" for="radio-stacked-3">Text message</label>
-    </div>
-  </fieldset>
+  {% call govukFieldset({
+    legendHtml: "<h2 class=\"govuk-heading-l\">How do you want to be contacted?</h2>"
+  }) %}
+    <span id="radio-stacked"></span>
+    {{ govukRadios({
+      "idPrefix": "radio-stacked",
+      "name": "radio-stacked",
+      "items": [
+        {
+          "value": "email",
+          "text": "Email"
+        },
+        {
+          "value": "phone",
+          "text": "Phone"
+        },
+        {
+          "value": "text",
+          "text": "Text message"
+        }
+      ]
+    }) }}
+  {% endcall %}
 
-  <fieldset class="govuk-c-fieldset" id="radio-inline">
-    <div class="govuk-c-radio govuk-c-radio--inline">
-      <input class="govuk-c-radio__input" id="radio-1" type="radio" name="radio-group" value="Yes">
-      <label class="govuk-c-radio__label" for="radio-1">Yes</label>
-    </div>
-    <div class="govuk-c-radio govuk-c-radio--inline">
-      <input class="govuk-c-radio__input" id="radio-2" type="radio" name="radio-group" value="No">
-      <label class="govuk-c-radio__label" for="radio-2">No</label>
-    </div>
-  </fieldset>
+  {% call govukFieldset() %}
```

Comment:
Do want want to use these components 'properly' by providing the fieldset options to the radio / checkbox macros, and including a legend?

Reply:
Nesting of components feels like the best way to put components together, feels intuitive to me anyway?

Reply:
I think I'd still prefer to use them in the way we've designed them, as in the first example [here](http://govuk-frontend-review.herokuapp.com/components/radios), but it's not a big issue.

Reply:
Will update, what decisions led to overloading the API? Feels better to keep the components distinct, as it's more like HTML?